plugins/optimization-detective/class-od-html-tag-processor.php

```diff
@@ -160,7 +179,32 @@ final class OD_HTML_Tag_Processor {
 	 * @param string $html HTML to process.
 	 */
 	public function __construct( string $html ) {
-		$this->processor = new WP_HTML_Tag_Processor( $html );
+
+		$this->processor = new class( $html ) extends WP_HTML_Tag_Processor {
+
+			/**
+			 * Appends HTML to the provided bookmark.
+			 *
+			 * @param string $bookmark Bookmark.
+			 * @param string $html     HTML to inject.
+			 * @return bool Whether the HTML was appended.
+			 */
+			public function append_html( string $bookmark, string $html ): bool {
+				if ( ! $this->has_bookmark( $bookmark ) ) {
+					return false;
+				}
+
+				$start = $this->bookmarks[ $bookmark ]->start;
+
+				$this->lexical_updates[] = new WP_HTML_Text_Replacement(
+					$start,
+					// In WordPress 6.5, the signature was changed from $end to $length.
+					version_compare( get_bloginfo( 'version' ), '6.5', '<' ) ? $start : 0,
```

Comment:
this probably doesn't matter much, but there's no need to recompute this on every call. the WordPress version won't change during the runtime (at least outside of the Playground where it could), so this could be set in the constructor or as a method-level static var.

Reply:
Good point. Although this will only be called twice at most, once for the head insertion and once for body. I'll add a static var.